tests/clients/test_docker_client.py

```diff
@@ -0,0 +1,270 @@
+
+    def test01_docker_get(self):
+        """
+        Test getting container.
+
+        **Test scenario**
+        #. Create docker container.
+        #. Get container ID using container name.
+        #. Try to get this container one time using container name.
+        #. Try to get a container with a non valid name, and make sure that it raises an error.
+        """
+        self.info("Get container ID using container name")
+        output, error = self.os_command("docker ps -aqf name={}".format(self.DOCKER_NAME))
+        self.assertTrue(output)
+
+        self.info("Try to get this container one time using container name")
+
+        docker = self.client.get(self.DOCKER_NAME)
+        self.assertIn(docker.short_id, output)
+
+        self.info("Try to get a container with a non valid name, and make sure that it raises an error")
+        with self.assertRaises(Exception) as error:
+            self.client.get("test_docker_{}".format(randint(100, 1000)))
+            self.assertTrue("No such container" in error.exception.args[0])
+
+    def test02_docker_list(self):
+        """
+        Test list docker.
+
+        **Test scenario**
+        #. Create the first docker.
+        #. Check that the docker is created correctly.
+        #. Get container ID using container name.
+        #. Create the second docker.
+        #. Try to list this container using list subcommand with option all=False to list only the running container.
+        #. Check the output of list command.
+        #. Use list subcommand with option all=True to list all containers.
+        #. Check the output of list command and make sure that it lists the two containers.
+        """
+        self.info("Check that the docker is created correctly")
+        self.assertTrue(self.docker_check_status(self.DOCKER_NAME, "true"))
+
+        self.info("Get container ID using container name")
+        DOCKER_ID_1 = self.client.get(self.DOCKER_NAME).short_id
+
+        self.info("Create second container")
+        DOCKER_NAME_2 = "docker_{}".format(randint(100, 10000))
+        DOCKER_ID_2 = self.client.create(DOCKER_NAME_2, self.DOCKER_IMAGE).short_id
+        self.assertTrue(self.docker_check_status(DOCKER_NAME_2, "false"))
+
+        self.info("Try to list container using list method with option all=False to list only the running container")
+        docker_list = self.client.list(all=False)
+
+        self.info("Check the output of list command")
+        self.info(" it should contain the first docker, but doesn't contain the second one")
+        self.assertIn(DOCKER_ID_1, str(docker_list))
+        self.assertNotIn(DOCKER_ID_2, str(docker_list))
+
+        self.info("Use list subcommand with option all=True to list all containers")
+        self.info("Check the output of list command and make sure that it lists the two containers")
+        docker_list = self.client.list(all=True)
+        self.assertIn(
+            DOCKER_ID_1 and DOCKER_ID_2,
```

Comment:
Adding two object with `and` result always in the second object, so this check only checking that DOCKER_ID_2 in the list.